src/byte_format/deserialize.cr

```diff
@@ -74,13 +74,13 @@ struct Crystalizer::ByteFormat
     Path.new deserialize(String, size)
   end
 
-  # Deserializes a `String` from reading from the `io`, delimited by a trailing `byte_delimiter`.
+  # Deserializes a `String` from reading from the `io`, delimited by a trailing `string_delimiter`.
   def deserialize(to type : String.class)
-    String.build do |str|
-      while (byte = @io.read_byte) && byte != @byte_delimiter
-        str.write_byte byte
-      end
-    end
+    if string_delimiter = @string_delimiter
+      @io.gets string_delimiter, true
+    else
+      @io.gets
+    end || ""
```

Comment:
This changes the default behavior, should be `gets_to_end`, otherwise the string delimiter is a new line. Then, the `|| ""` can be move up to the `@io.gets string_delimiter` line.

Reply:
Mm right, will update, thanks!